### PR TITLE
FIRApp: add apple platform to firebaseUserAgent

### PR DESCRIPTION
**Pull request by maksymmalyhin**
Collect stats on Firebase usage by Apple platform (b/146507104).

### CODE REVIEW

File: FirebaseCore/Sources/FIRApp.m [hunk 1]
```diff
@@ -606,6 +610,32 @@ + (BOOL)hasSwiftRuntime {
...
+#if TARGET_OS_OSX
+  applePlatform = @"macos";
+#endif  // TARGET_OS_OSX
+
+#if TARGET_OS_WATCH
+  applePlatform = @"watchos";
+#endif  // TARGET_OS_WATCH
+
+#if TARGET_OS_MACCATALYST
+  applePlatform = @"maccatalyst";
```
Comment (paulb777) [inline FirebaseCore/Sources/FIRApp.m]: Please check that this define is not combined with any of the others.  We might want it first with `#elif`s

Comment (maksymmalyhin) [reply FirebaseCore/Sources/FIRApp.m]: According to the current API docs it should not. I wonder if we can test it on Travis/GHA somehow? 
Screen Shot 2020-02-19 at 3 30 48 PM

Comment (maksymmalyhin) [reply FirebaseCore/Sources/FIRApp.m]: Ok, for Catalyst app run on macOS both `TARGET_OS_IOS` and `TARGET_OS_MACCATALYST` are set. I wonder what platform we would like to log in this case? I assume `maccatalyst`. WDYT?

Comment (ryanwilson) [reply FirebaseCore/Sources/FIRApp.m]: I would say `maccatalyst`.

Comment (maksymmalyhin) [reply FirebaseCore/Sources/FIRApp.m]: Updated with `#elif` accordingly.

File: FirebaseCore/Sources/FIRApp.m [hunk 2]
```diff
@@ -606,6 +610,24 @@ + (BOOL)hasSwiftRuntime {
   return hasSwiftRuntime;
 }
 
++ (NSString *)applePlatform {
+  NSString *applePlatform = @"unknown";
+
+#if TARGET_OS_MACCATALYST
```
Comment (paulb777) [inline FirebaseCore/Sources/FIRApp.m]: Add a comment summarizing the discussion so we remember this is order-sensitive

Comment (maksymmalyhin) [reply FirebaseCore/Sources/FIRApp.m]: Added the comment. Is it what you meant?

Comment (paulb777) [reply FirebaseCore/Sources/FIRApp.m]: Yep. Thanks!

**Pull request merged**